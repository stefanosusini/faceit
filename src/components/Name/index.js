import React, { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import utilsStyles from '../../scss/utils.module.scss';

import styles from './index.module.scss';

let Name = ({ className, name, loading }) => {
  return (
    <div className={classNames(
      styles.container,
      { [utilsStyles.placeholderBackground]: loading },
      className
    )}>
      <p className={classNames(
        styles.text,
        { [utilsStyles.placeholderBackground]: loading }
      )}>{!loading && name}</p>
    </div>
  );
};

Name = memo(Name);

Name.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  loading: PropTypes.bool,
};
Name.defaultProps = {
  loading: false,
};

export default Name;
