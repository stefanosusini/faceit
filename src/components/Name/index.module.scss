@import "scss/mixins";

.container {
  display: flex;
  align-items: center;
  overflow: hidden;
  padding: 0 0.8rem 0 1rem;
  flex-grow: 1;
  @include responsiveRowHeight();
}

.text {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 1.2em;
  letter-spacing: 0.5px;
  margin: 0;
}
