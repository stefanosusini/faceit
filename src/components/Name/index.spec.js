import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Name from './index';

describe('Name component', () => {
  it('should render', () => {
    const wrapper = shallow(<Name name="NAME" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  describe('if loading', () => {
    it('should render the loading state', () => {
      const wrapper = shallow(<Name name="NAME" loading={true} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  describe('if className is provided', () => {
    it('should render the component with the class name', () => {
      const wrapper = shallow(<Name name="NAME" className="CLASSNAME" />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    describe('if loading', () => {
      it('should render the class names in the correct order', () => {
        const wrapper = shallow(<Name name="NAME" className="CLASSNAME" loading={true} />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
