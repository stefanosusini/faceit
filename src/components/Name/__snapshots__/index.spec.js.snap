// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name component if className is provided if loading should render the class names in the correct order 1`] = `
<div
  className="container placeholderBackground CLASSNAME"
>
  <p
    className="text placeholderBackground"
  />
</div>
`;

exports[`Name component if className is provided should render the component with the class name 1`] = `
<div
  className="container CLASSNAME"
>
  <p
    className="text"
  >
    NAME
  </p>
</div>
`;

exports[`Name component if loading should render the loading state 1`] = `
<div
  className="container placeholderBackground"
>
  <p
    className="text placeholderBackground"
  />
</div>
`;

exports[`Name component should render 1`] = `
<div
  className="container"
>
  <p
    className="text"
  >
    NAME
  </p>
</div>
`;
